### PR TITLE
Fix Doc URLs, add user-agent

### PR DESCRIPTION
**Pull request by akr4s1a**
Hi there, I saw this project in Modtoberfest and saw I could add a few small things.

- I've changed the docs URLs to remove api-spec/ so they resolve 
- I added a header using the supplied slug. 
- I wanted to contribute a mechanism to take the rate limit supplied by the response in https://docs.modrinth.com/#section/Ratelimits 

but without a local testing environment I didn't feel comfortable leaving the rate limit mechanism in. It will need a way to send the request again, sendRequest() could be encapsulated and moved

### CODE REVIEW

**Comment by funnyboy-roks** [comment]
Thank you for your PR!

- Thank you for updating the docs urls
- For the user agent, we should be more specific.  Something like: `<project-id> via funnyboy-roks/modrinth-auto-desc` would work for me.
- For the rate limit, I think this is a good idea to implement, though I don't have a good setup for testing this at the moment.  Would you be willing to put that in a second PR, so that we can merge the other two things in this PR and look at that in the future?

**Comment by akr4s1a** [comment]
Sure I can split off the rate limiting and open a draft PR for that.

For getting project-id for a better user-agent, what do you think the best way to go about it is? I think adding optional entries in front matter could be good for getting contact info and to specify the project.

**Comment by funnyboy-roks** [comment]
We can add those as optional arguments for the action.  See the [action.yml](https://github.com/funnyboy-roks/modrinth-auto-desc/blob/main/action.yml) file, so add a `project-name` field and if set, use `<project-name> (<slug>) via funnyboy-roks/modrinth-auto-desc` and if not set then just `<slug> via funnyboy-roks/modrinth-auto-desc`.

I do want to keep the slug in the user agent, in case the user sets a bad project name.

**Comment by akr4s1a** [comment]
Something like this?

File: action.yml [hunk 1]
```diff
@@ -12,6 +12,10 @@ inputs:
     description: 'The path to the file to use for the description'
     required: false
     default: 'README.md'
+  project-name:
+    description: 'Used in the user-agent for identification purposes'
+    required: false
+    default: 'REPLACEME'
```
Comment (funnyboy-roks) [inline action.yml:18]: This is good, would prefer the default to be something more like `__unset`

File: index.js [hunk 2]
```diff
@@ -12,6 +12,10 @@ const main = async () => {
         // Grab the slug if it's a url
         slug = slug.substring(slug.lastIndexOf('/') + 1);
 
+        //project-name for the user agent
+        let user_agent = actions.getInput("project-name")
+        user_agent = user_agent == "REPLACEME" ? slug : `${user_agent} (${slug})`
```
Comment (funnyboy-roks) [inline index.js:17]: Very nitpicky, but can you use single quotes (`'`) instead of the double quotes (`"`)

**Review by funnyboy-roks — CHANGES_REQUESTED** [review]
Looks good, just minor issues

**Review by funnyboy-roks — APPROVED** [review]
Looks good!

**Pull request merged**